### PR TITLE
mirage-no-xen etc...

### PR DESCRIPTION
**Pull request by hannesm**
* add mirage-no-xen
* rename gmp to gmp-xen (and build only xen part)
* optionally depend on gmp-xen in zarith if mirage-xen is installed

completely untested since no xen machine at my hands...

### CODE REVIEW

File: packages/zarith/zarith.1.3/opam [hunk 1]
```diff
@@ -4,6 +4,7 @@ patches: ["xen_linkopts.patch"]
 build: ["./mirage-install.sh"]
 remove: [["ocamlfind" "remove" "zarith"]]
 depends: [
-  "gmp"
+  ("mirage-no-xen" | ("mirage-xen" & "gmp-xen"))
```
Comment (samoht) [inline packages/zarith/zarith.1.3/opam]: should be `("mirage-no-xen" & "conf-gmp") | ("mirage-xen" & "gmp-xen")` I think

File: packages/zarith/zarith.1.3/files/xen_linkopts.patch [hunk 2]
```diff
@@ -2,10 +2,7 @@ diff --git a/META b/META
 index c082c65..14394d1 100644
 --- a/META
 +++ b/META
-@@ -1,5 +1,6 @@
- description = "Arbitrary precision integers"
--requires = ""
-+requires = "gmp"
+@@ -3,3 +3,4 @@
```
Comment (samoht) [inline packages/zarith/zarith.1.3/files/xen_linkopts.patch]: Not sure how to get the dependency`gmp-xen` back here...

File: packages/zarith/zarith.1.3/opam (hunk 1 above)
Comment (hannesm) [inline packages/zarith/zarith.1.3/opam]: no, we always want conf-gmp! this is from old zarith opam...

File: packages/zarith/zarith.1.3/files/xen_linkopts.patch (hunk 2 above)
Comment (hannesm) [inline packages/zarith/zarith.1.3/files/xen_linkopts.patch]: maybe a patch in the mirage-install.sh?

Comment (samoht) [inline packages/zarith/zarith.1.3/files/xen_linkopts.patch]: I think something along the lines of https://github.com/mirage/mirage-dev/pull/56/files#diff-0eca2e8236c2b30dae11dde168158c5aR10 would work. ie. define sub-package `zarith.xen` which depends on `gmp-xen`. The only think is that then `zarith.xen` need to be pulled at the right place when needed...

Comment (samoht) [inline packages/zarith/zarith.1.3/files/xen_linkopts.patch]: Or maybe requiring an optionnally existing ocamlfind package works... I'll check that tomorrow on my xen box.

**Pull request merged**